lms/djangoapps/courseware/masquerade.py

```diff
@@ -116,3 +174,146 @@ def get_masquerading_group_info(user, course_key):
     if not course_masquerade:
         return None, None
     return course_masquerade.group_id, course_masquerade.user_partition_id
+
+
+# Sentinel object to mark deleted objects in the session cache
+_deleted_sentinel = object()
+
+
+class MasqueradingFieldDataCache(object):
+    """
+    A wrapper for FieldDataCache used when masquerading as a different user.  Data is read from the
+    underlying cache, but written to the session, so the state of the actual user is not changed.
+    """
```

Comment:
It would be better to make a `MasqueradingFieldData`, rather than a `MasqueradingFieldDataCache`. Then it would be more applicable to XBlocks, rather than the lms specific machinery.